Enable LCD_PAR_S035 shield 8080 mode for RT700

KATE-WANG-NXP opened the pull request:
1.Fix wrong handling of color coding in NXP DCnano DBI driver and dsi_mcux_2l driver
2.Update st7796s display driver to support color coding
3.Enable shield LCD_PAR_S035 8080 mode on RT700

KATE-WANG-NXP committed:
drivers: mipi_dsi: Fix bugs in dsi_mcux_2l

The color_coding in mipi_dbi_config shall not be combined with
bus_type. They shal have their seperate definitions in binding
to avoid confusion.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
drivers: display: st7796s: Update driver

Add the support of DBI bus color coding configuration, also
pass down height/width/pitch of the frmae buffer descriptor
during frame update. Some DBI controller may need this configuration.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
samples: drivers: display: Add MIMXRT700 EVK with LCD-PAR-S035 shield test

Add test configuration for display sample using MIMXRT700 EVK
board with LCD-PAR-S035 shield in 8080 parallel interface mode.

This enables testing of the display driver sample with the newly
supported MIMXRT700 EVK and LCD-PAR-S035 shield combination.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
drivers: mipi_dbi: Fix bugs in NXP DCnano DBI driver

1.The color_coding in mipi_dbi_config shall not be extracting it from
the bus_type field and shall have dedicated color_coding field.
2.Update device tree binding to include pinctrl support and
declare mipi-dbi bus compatibility.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

JarmouniA reviewed (dismissed):
Please submit the proposed DBI color-coding changes in a separate PR that contains only the strictly required changes.

KATE-WANG-NXP commented:
Rebase and reopen the PR since the depended PR has been merged

zejiang0jason commented on boards/shields/lcd_par_s035/boards/mimxrt700_evk_mimxrt798s_cm33_cpu0.conf:
```diff
@@ -0,0 +1 @@
+CONFIG_DCACHE=n # DCNano DBi controller cannot be used with data cahce
```
DBi -> DBI

zejiang0jason commented on boards/shields/lcd_par_s035/boards/mimxrt700_evk_mimxrt798s_cm33_cpu0.conf:
```diff
@@ -0,0 +1 @@
+CONFIG_DCACHE=n # DCNano DBi controller cannot be used with data cahce
```
Could runtime maintain the cache or place frame memory in non-cacheable region?

KATE-WANG-NXP replied:
I prefer this ways too but currently I believe no display/video related example support this feature and I think maybe this update should be done in different PR?

KATE-WANG-NXP replied:
Unfortunately this does not work for this case NCNano cannot work with cache on RT700

yvesll commented on boards/nxp/mimxrt700_evk/mimxrt700_evk_mimxrt798s_cm33_cpu0.dts:
```diff
@@ -296,6 +309,14 @@ zephyr_mipi_dsi: &mipi_dsi {};
 zephyr_lcdif: &lcdif {};
 
+zephyr_mipi_dbi_parallel: &lcdif {
+	reset-gpios = <&gpio2 15 GPIO_ACTIVE_HIGH>;
```
Need add `reset-gpios` for `nxp,dcnano-lcdif`

KATE-WANG-NXP commented:
Hi all, please help review this PR, many thanks

KATE-WANG-NXP committed:
doc: boards: nxp: Add new supported panels for RT700

Add new supported panel LCD_PAR_S035 (8080 mode)
in board doc.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
boards: shields: lcd_par_s035: Add MIMXRT700 EVK support

Add board-specific configuration for LCD-PAR-S035 shield on
MIMXRT700 EVK board. This includes:

- Device tree overlay configuring LCDIF controller with timing
  parameters for 8080 parallel interface (write periods, assert/
  deassert timings, chip select timing)
- ST7796S display controller configuration with 8-bit 8080 bus
  mode and RGB565 color coding
- Board configuration disabling data cache due to DCNano DBI
  controller incompatibility

The configuration enables the LCD-PAR-S035 shield to work with
the MIMXRT700 EVK's MIPI DBI parallel interface.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
boards: mimxrt700_evk: Add LCD 8080 parallel interface support

Add pinctrl and device tree configuration for LCD 8080 parallel
interface on MIMXRT700 EVK board. This includes:

- Pinctrl configuration for LCDIF DBI interface with 8-bit data
  bus (PIO2_6 to PIO2_13), control signals (CSX, WRX, DCX, RWDX),
  and GPIO for reset
- LCD 8080 connector node
- Aliases for MIPI DBI parallel interface and touch panel I2C

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>
